File: uploaddemo/src/business/galary/java/com/jw/galary/img/util/CornerUtils.java


package com.jw.galary.img.util;

import android.graphics.drawable.ColorDrawable;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.GradientDrawable;
import android.graphics.drawable.StateListDrawable;

public class CornerUtils {
    public CornerUtils() {
    }

    public static Drawable cornerDrawable(int bgColor, float cornerradius) {
        GradientDrawable bg = new GradientDrawable();
        bg.setCornerRadius(cornerradius);
        bg.setColor(bgColor);
        return bg;
    }

    public static Drawable cornerDrawable(int bgColor, float[] cornerradius) {
        GradientDrawable bg = new GradientDrawable();
        bg.setCornerRadii(cornerradius);
        bg.setColor(bgColor);
        return bg;
    }

    public static Drawable cornerDrawable(int bgColor, float[] cornerradius, int borderwidth, int bordercolor) {
        GradientDrawable bg = new GradientDrawable();
        bg.setCornerRadii(cornerradius);
        bg.setStroke(borderwidth, bordercolor);
        bg.setColor(bgColor);
        return bg;
    }

    public static StateListDrawable btnSelector(float radius, int normalColor, int pressColor, int enableColor, int position) {
        StateListDrawable bg = new StateListDrawable();
        Drawable normal = null;
        Drawable pressed = null;
        Drawable enabled = null;
        if (position == 0) {
            normal = cornerDrawable(normalColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, radius, radius});
            pressed = cornerDrawable(pressColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, radius, radius});
            enabled = cornerDrawable(enableColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, 0.0F, 0.0F, radius, radius});
        } else if (position == 1) {
            normal = cornerDrawable(normalColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, radius, radius, 0.0F, 0.0F});
            pressed = cornerDrawable(pressColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, radius, radius, 0.0F, 0.0F});
            enabled = cornerDrawable(enableColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, radius, radius, 0.0F, 0.0F});
        } else if (position == -1) {
            normal = cornerDrawable(normalColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, radius, radius, radius, radius});
            pressed = cornerDrawable(pressColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, radius, radius, radius, radius});
            enabled = cornerDrawable(enableColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, radius, radius, radius, radius});
        } else if (position == -2) {
            normal = cornerDrawable(normalColor, radius);
            pressed = cornerDrawable(pressColor, radius);
            enabled = cornerDrawable(enableColor, radius);
        }

        bg.addState(new int[]{-16842919}, normal);
        bg.addState(new int[]{16842919}, pressed);
        bg.addState(new int[]{16842910}, enabled);
        return bg;
    }

    public static StateListDrawable listItemSelector(float radius, int normalColor, int pressColor, boolean isLastPostion) {
        StateListDrawable bg = new StateListDrawable();
        Drawable normal = null;
        Drawable pressed = null;
        if (!isLastPostion) {
            normal = new ColorDrawable(normalColor);
            pressed = new ColorDrawable(pressColor);
        } else {
            normal = cornerDrawable(normalColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, radius, radius, radius, radius});
            pressed = cornerDrawable(pressColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, radius, radius, radius, radius});
        }

        bg.addState(new int[]{-16842919}, (Drawable)normal);
        bg.addState(new int[]{16842919}, (Drawable)pressed);
        return bg;
    }

    public static StateListDrawable listItemSelector(float radius, int normalColor, int pressColor, int itemTotalSize, int itemPosition) {
        StateListDrawable bg = new StateListDrawable();
        Drawable normal = null;
        Drawable pressed = null;
        if (itemPosition == 0 && itemPosition == itemTotalSize - 1) {
            normal = cornerDrawable(normalColor, new float[]{radius, radius, radius, radius, radius, radius, radius, radius});
            pressed = cornerDrawable(pressColor, new float[]{radius, radius, radius, radius, radius, radius, radius, radius});
        } else if (itemPosition == 0) {
            normal = cornerDrawable(normalColor, new float[]{radius, radius, radius, radius, 0.0F, 0.0F, 0.0F, 0.0F});
            pressed = cornerDrawable(pressColor, new float[]{radius, radius, radius, radius, 0.0F, 0.0F, 0.0F, 0.0F});
        } else if (itemPosition == itemTotalSize - 1) {
            normal = cornerDrawable(normalColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, radius, radius, radius, radius});
            pressed = cornerDrawable(pressColor, new float[]{0.0F, 0.0F, 0.0F, 0.0F, radius, radius, radius, radius});
        } else {
            normal = new ColorDrawable(normalColor);
            pressed = new ColorDrawable(pressColor);
        }

        bg.addState(new int[]{-16842919}, (Drawable)normal);
        bg.addState(new int[]{16842919}, (Drawable)pressed);
        return bg;
    }
}

File: library/src/main/java/com/jw/library/utils/CornerUtils.kt
package com.jw.library.utils

import android.graphics.drawable.ColorDrawable
import android.graphics.drawable.Drawable
import android.graphics.drawable.GradientDrawable
import android.graphics.drawable.StateListDrawable

object CornerUtils {

    fun cornerDrawable(bgColor: Int, cornerradius: Float): Drawable {
        val bg = GradientDrawable()
        bg.cornerRadius = cornerradius
        bg.setColor(bgColor)
        return bg
    }

    fun cornerDrawable(bgColor: Int, cornerradius: FloatArray): Drawable {
        val bg = GradientDrawable()
        bg.cornerRadii = cornerradius
        bg.setColor(bgColor)
        return bg
    }

    fun cornerDrawable(
        bgColor: Int,
        cornerradius: FloatArray,
        borderwidth: Int,
        bordercolor: Int
    ): Drawable {
        val bg = GradientDrawable()
        bg.cornerRadii = cornerradius
        bg.setStroke(borderwidth, bordercolor)
        bg.setColor(bgColor)
        return bg
    }

    fun btnSelector(
        radius: Float,
        normalColor: Int,
        pressColor: Int,
        enableColor: Int,
        position: Int
    ): StateListDrawable {
        val bg = StateListDrawable()
        var normal: Drawable? = null
        var pressed: Drawable? = null
        var enabled: Drawable? = null
        if (position == 0) {
            normal = cornerDrawable(
                normalColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, radius, radius)
            )
            pressed = cornerDrawable(
                pressColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, radius, radius)
            )
            enabled = cornerDrawable(
                enableColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, radius, radius)
            )
        } else if (position == 1) {
            normal = cornerDrawable(
                normalColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, radius, radius, 0.0f, 0.0f)
            )
            pressed = cornerDrawable(
                pressColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, radius, radius, 0.0f, 0.0f)
            )
            enabled = cornerDrawable(
                enableColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, radius, radius, 0.0f, 0.0f)
            )
        } else if (position == -1) {
            normal = cornerDrawable(
                normalColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, radius, radius, radius, radius)
            )
            pressed = cornerDrawable(
                pressColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, radius, radius, radius, radius)
            )
            enabled = cornerDrawable(
                enableColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, radius, radius, radius, radius)
            )
        } else if (position == -2) {
            normal = cornerDrawable(normalColor, radius)
            pressed = cornerDrawable(pressColor, radius)
            enabled = cornerDrawable(enableColor, radius)
        }

        bg.addState(intArrayOf(-16842919), normal)
        bg.addState(intArrayOf(16842919), pressed)
        bg.addState(intArrayOf(16842910), enabled)
        return bg
    }

    fun listItemSelector(
        radius: Float,
        normalColor: Int,
        pressColor: Int,
        isLastPostion: Boolean
    ): StateListDrawable {
        val bg = StateListDrawable()
        var normal: Drawable? = null
        var pressed: Drawable? = null
        if (!isLastPostion) {
            normal = ColorDrawable(normalColor)
            pressed = ColorDrawable(pressColor)
        } else {
            normal = cornerDrawable(
                normalColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, radius, radius, radius, radius)
            )
            pressed = cornerDrawable(
                pressColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, radius, radius, radius, radius)
            )
        }

        bg.addState(intArrayOf(-16842919), normal as Drawable?)
        bg.addState(intArrayOf(16842919), pressed as Drawable?)
        return bg
    }

    fun listItemSelector(
        radius: Float,
        normalColor: Int,
        pressColor: Int,
        itemTotalSize: Int,
        itemPosition: Int
    ): StateListDrawable {
        val bg = StateListDrawable()
        var normal: Drawable? = null
        var pressed: Drawable? = null
        if (itemPosition == 0 && itemPosition == itemTotalSize - 1) {
            normal = cornerDrawable(
                normalColor,
                floatArrayOf(radius, radius, radius, radius, radius, radius, radius, radius)
            )
            pressed = cornerDrawable(
                pressColor,
                floatArrayOf(radius, radius, radius, radius, radius, radius, radius, radius)
            )
        } else if (itemPosition == 0) {
            normal = cornerDrawable(
                normalColor,
                floatArrayOf(radius, radius, radius, radius, 0.0f, 0.0f, 0.0f, 0.0f)
            )
            pressed = cornerDrawable(
                pressColor,
                floatArrayOf(radius, radius, radius, radius, 0.0f, 0.0f, 0.0f, 0.0f)
            )
        } else if (itemPosition == itemTotalSize - 1) {
            normal = cornerDrawable(
                normalColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, radius, radius, radius, radius)
            )
            pressed = cornerDrawable(
                pressColor,
                floatArrayOf(0.0f, 0.0f, 0.0f, 0.0f, radius, radius, radius, radius)
            )
        } else {
            normal = ColorDrawable(normalColor)
            pressed = ColorDrawable(pressColor)
        }

        bg.addState(intArrayOf(-16842919), normal as Drawable?)
        bg.addState(intArrayOf(16842919), pressed as Drawable?)
        return bg
    }
}
